### DISCUSSION TITLE
Set correct types on persisted slice to get autocompletion for persist's methods

### DISCUSSION THREAD

**Discussion by diego20** [pmndrs/zustand#2086] [Q&A] [answered]
I have the following store in which I am only persisting the first slice. The store is working correctly.

``` ts
const useBoundStore = create<BrandSlice & HelpCenterSlice & ErrorSlice>()(
  (...a) => ({
    ...persist(createBrandSlice, {
      name: 'account-data',
      partialize: (state) => ({ selectedAccount: state.selectedAccount }),
    })(...a),
    ...createHelpCenterSlice(...a),
    ...createErrorSlice(...a),
  })
);
```
I want to clear localStorage this way `useBoundStore.persist.clearStorage()`, following the docs, and I know the method exists and works because I use it.
**The only problem is I am not getting TS autocompletion** and I'm seeing the red squiggle under `persist`

What is the correct way to type the store in this case to make TS know `useBoundStore` has an object `persist`?

**Comment by dai-shi** [comment, marked as answer]
I guess it's never designed to be used like for a slice. As you use `partialize` option, `persist` the entire state would work.
